airbyte-cdk/java/airbyte-cdk/db-sources/src/main/kotlin/io/airbyte/cdk/components/debezium/StateFilesAccessor.kt

```diff
@@ -0,0 +1,192 @@
+                STORE_RECORD_METHOD.invoke(fileSchemaHistory, r)
+            } catch (e: IllegalAccessException) {
+                throw RuntimeException(e)
+            } catch (e: InvocationTargetException) {
+                throw RuntimeException(e)
+            }
+        }
+        fileSchemaHistory.stop()
+    }
+
+    init {
+        try {
+            workingDir = Files.createTempDirectory(Path.of("/tmp"), "airbyte-debezium-state")
+        } catch (e: IOException) {
+            throw UncheckedIOException(e)
+        }
+        offsetFilePath = workingDir.resolve("offset.dat")
+        schemaFilePath = workingDir.resolve("dbhistory.dat")
+        // Create and configure FileOffsetBackingStore instance.
+        fileOffsetBackingStore = FileOffsetBackingStore(keyConverter())
+        val offsetProps = HashMap<String, String>()
+        offsetProps[WorkerConfig.KEY_CONVERTER_CLASS_CONFIG] = JsonConverter::class.java.name
+        offsetProps[WorkerConfig.VALUE_CONVERTER_CLASS_CONFIG] = JsonConverter::class.java.name
+        offsetProps[StandaloneConfig.OFFSET_STORAGE_FILE_FILENAME_CONFIG] =
+            offsetFilePath.toString()
+        fileOffsetBackingStore.configure(StandaloneConfig(offsetProps))
+        // Create and configure FileSchemaHistory instance.
+        fileSchemaHistory = FileSchemaHistory()
+        val schemaConfig =
+            Configuration.create()
+                .with(FileSchemaHistory.FILE_PATH, schemaFilePath.toString())
+                .build()
+        fileSchemaHistory.configure(
+            schemaConfig,
+            HistoryRecordComparator.INSTANCE,
+            SchemaHistoryListener.NOOP,
+            false
+        )
+    }
+
+    companion object {
+        // Use reflection to access the necessary protected methods in FileSchemaHistory.
+        private val STORE_RECORD_METHOD: Method
+        private val RECOVER_RECORDS_METHOD: Method
+
+        init {
+            try {
+                STORE_RECORD_METHOD =
+                    AbstractFileBasedSchemaHistory::class
+                        .java
+                        .getDeclaredMethod("storeRecord", HistoryRecord::class.java)
+                RECOVER_RECORDS_METHOD =
+                    AbstractFileBasedSchemaHistory::class
+                        .java
+                        .getDeclaredMethod("recoverRecords", Consumer::class.java)
+            } catch (e: NoSuchMethodException) {
+                throw RuntimeException(e)
+            }
+            STORE_RECORD_METHOD.setAccessible(true)
+            RECOVER_RECORDS_METHOD.setAccessible(true)
+        }
+
+        private fun keyConverter(): Converter {
+            val converter = JsonConverter()
+            val converterConfig =
+                mapOf(Pair(JsonConverterConfig.SCHEMAS_ENABLE_CONFIG, false.toString()))
+            converter.configure(converterConfig, true)
+            return converter
+        }
+
+        private fun toJson(byteBuffer: ByteBuffer): JsonNode {
```

Comment:
are those 2 ever used?

Reply:
my bad, for some reason, search didn't find it

Reply:
No worries, it's possible that there's detritus lying around, always good to check.